### ISSUE TITLE
Create plugin for nilearn

### ISSUE THREAD

**Issue by mnarayan** [skggm/skggm#81]
Since our estimators are sklearn compatible, we can enable those who use nilearn for graphical model estimation to use skggm estimators as well. 

This would involve using [ConnectivityMeasure](http://nilearn.github.io/modules/generated/nilearn.connectome.ConnectivityMeasure.html#examples-using-nilearn-connectome-connectivitymeasure) from nilearn as a base estimator class (which is designed for dealing with multiple graphical models) and then creating an skggm relevant implementation under the hood.  Would be useful for application examples, where a similar procedure would need to be applied in any case.

**Comment by jasonlaska** [comment]
This seems like just an example right?  How is this different than just showing an examples that includes lines similar to:

```
from nilearn.connectome import ConnectivityMeasure

cm = ConnectivityMeasure(
    cov_estimator=QuicGraphLassoCV, kind='precision'
)

cm.fit(...)
```

**Comment by mnarayan** [comment]
This is definitely an example not a core feature.

But it needs to handle some issues relating to dealing with multiple
subjects/networks. The expectation of the output is that sparse support
will be comparable across all of them in some sense.

Nilearn folks discourage using this base estimator for single graphs. But
as you point out. It is trivial for this naive usecase where we are
compatible.

**Comment by dpisner453** [comment]
@mnarayan -- this was an interesting discussion that I was hoping we could resume?

My two cents on this is that the nilearn folks are correct that _single_ precision matrices for single individuals are not comparable across individuals (i.e. where skggm is compatible). The philosophy behind PyNets, however, is that the 'intersection' across a distribution of precision matrices built upon multiple node definitions for single individuals _are_ comparable across individuals, where skggm is compatible. 

Curious to hear your thoughts on this, and particularly whether you think this impacts how skggm's estimators should be used in such scenarios...

-derek

**Comment by mnarayan** [comment]
Hi @dpisner453. I think the issue of whether individual graphs are comparable or not depends on 
a) The kinds of individual level and population level statistical models assumed 
b) The types of loss functions used to compare precision matrices.  
c) There are also many flexible intermediate options that are not yet implemented here to optimally pool information across subjects without compromising individual variation. This is ongoing research where there are many open questions and skggm is avenue to benchmark and test these options.
Our initial focus has been on building better default estimators at the level of a single data matrix, not that no pooling ought to happen. We don't have anything implementing estimators for a population of networks yet. 

Last I checked nilearn's implementations prefer to assume shared model selection across all individuals and refits that model onto individual subject data.  This is quite reasonable for the sorts of applications nilearn is used for like classification/clustering and so forth. 

I am familiar with nilearn's ConnectivityMeasure API. Does PyNets have something similar as well? If PyNets has an explicit separate subject estimation philosophy, then our current estimators will work easily for you out of the box.

**Comment by dpisner453** [comment]
Thanks for the response @mnarayan !

Are you referring to population-based network estimators like [D-trace loss](https://academic.oup.com/biomet/article-abstract/104/4/755/4460032)?  

I should clarify I am not suggesting the use of estimators that pool information across subjects (i.e. inter-subject models), only those that pool information _within_ subjects, because, as you noted, PyNets has an explicit separate subject estimation philosophy...

_Our initial focus has been on building better default estimators at the level of a single data matrix, not that no pooling ought to happen. We don't have anything implementing estimators for a population of networks yet._

Fair. And it's not entirely clear, in the the context of structural and functional brain networks at least, the extent to which any such pooling would be better accomplished at the level of individual matrix estimators vs. at the level of the actual stats calculated from those matrices, or both.

_Last I checked nilearn's implementations prefer to assume shared model selection across all individuals and refits that model onto individual subject data. This is quite reasonable for the sorts of applications nilearn is used for like classification/clustering and so forth._
Definitely, and I've used Nilearn in precisely that manner in the past. But I'm in the business of studying individual network neurophenotypes where shared model selection can obscure nuanced individual differences in functional networks in particular. To be able to study brain networks in this way means going beyond black-box approaches like classification (i.e. also extracting robust network statistics that can be used with GLM, SEM, etc.)

PyNets does have the ability to access nilearn's ConnectivityMeasure API library of estimators but  is also able, as you noted, to use skggm's estimators out-of-the-box. Very much looking forward to seeing what skggm comes up with next.

-derek
